app/src/main/java/com/github/scribeWizTeam/scribewiz/models/BadgeModel.kt

```diff
@@ -0,0 +1,109 @@
+package com.github.scribeWizTeam.scribewiz.models
+
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.toObject
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.tasks.await
+import java.text.DateFormat
+import java.util.*
+
+enum class BadgeRanks{
+    GOLD,
+    SILVER,
+    BRONZE
+}
+data class BadgeModel(override var id: String? = "",
+                      var badgeName: String? = "",
+                      var challengeID: String? = "",
+                      var dateObtained: String? = "",
+                      var photoID: String? = "",
+                      var rank: Int? = 0
+                      )
+: Model{
+
+    companion object Controller {
+        const val COLLECTION = "Badges"
+
+        /**
+         * Returns a set containing the badges belonging to the user.
+         * To get the user, LocalContext.current can be used
+         */
+        fun getAllBadgesFromUser(user: UserModel) : MutableSet<BadgeModel> {
+            val db = Firebase.firestore
+
+            val badges : MutableSet<BadgeModel> = mutableSetOf()
+
+            runBlocking {
+                val job = launch {
+                    for (id in user.badges!!) {
+                        db.collection("Users")
+                            .document(user.id!!)
+                            .collection(COLLECTION)
+                            .document(id)
+                            .get()
+                            .await()
+                            .toObject<BadgeModel>()?.let {
+                                badges.add(it)
+                            }
+                    }
+                }
+                job.join()
+            }
+            return badges
+        }
+
+        /**
+         * Adds the given badge to the provided user.
+         * If no badge is provided, a default test badge will be added
+         * To get the user, LocalContext.current can be used
+         */
+        fun addBadgeToUser(user: UserModel, badge: BadgeModel?){
+            val time = Calendar.getInstance().time
+            val formatter = DateFormat.getDateInstance()
+
+            var badgeData = BadgeModel(
```

Comment:
you can put it as a default value in the parameters of the function instead, 
Like:
```
fun addBadgeToUser(user: UserModel, badge: BadgeModel = BadgeModel(...)) {
...
```